x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -776,6 +776,7 @@ Collection<Object> createComponents(
             client,
             systemIndices.getProfileIndexManager(),
             clusterService,
+            realms::getDomainConfig,
```

Comment:
I'd like to refactor Domains into its own class in future. So for now I am passing a method reference here instead of `Realms` because I don't think `ProfileService` needs to know `Realms` which is quite heavy.

Reply:
Personally, this doesn't bother me.